<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1021</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus Data Fetcher</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../data/qmodbus.qrc">
    <normaloff>:/images/logo.png</normaloff>:/images/logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>3</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Modbus RTU</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="RtuSettingsWidget" name="rtuSettingsWidget" native="true"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Modbus TCP</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="TcpIpSettingsWidget" name="tcpSettingsWidget" native="true"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Modbus ASCII</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="sizeConstraint">
           <enum>QLayout::SetNoConstraint</enum>
          </property>
          <item>
           <widget class="AsciiSettingsWidget" name="asciiSettingsWidget" native="true"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_4">
         <property name="mouseTracking">
          <bool>true</bool>
         </property>
         <attribute name="title">
          <string>DB</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetNoConstraint</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="dbForm" name="dbFormWidget" native="true"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>ModBus Request</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="topMargin">
          <number>3</number>
         </property>
         <item row="4" column="0" colspan="4">
          <widget class="QTextEdit" name="requestPreview">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>300</width>
             <height>25</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>Courier</family>
            </font>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="undoRedoEnabled">
            <bool>false</bool>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
         <item row="5" column="4">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>Press Ctrl for polling mode...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QSpinBox" name="numCoils">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item row="4" column="4">
          <widget class="QPushButton" name="sendBtn">
           <property name="text">
            <string>Send</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="7">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="5" rowspan="2" colspan="2">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Polling interval, s.</string>
           </property>
          </widget>
         </item>
         <item row="3" column="4" colspan="3">
          <widget class="QCheckBox" name="checkBoxHexData">
           <property name="text">
            <string>Display hex data</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2" rowspan="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Function code</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" rowspan="2">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Slave ID</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="3" rowspan="2">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Start address</string>
           </property>
          </widget>
         </item>
         <item row="2" column="5" colspan="2">
          <widget class="QSpinBox" name="pollingInterval">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="0" column="4" rowspan="2">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Num of coils</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QComboBox" name="functionCode">
           <item>
            <property name="text">
             <string>Read Holding Registers (0x03)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Read Coils (0x01)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Read Discrete Inputs (0x02)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Read Input Registers (0x04)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Write Single Coil (0x05)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Write Single Register (0x06)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Write Multiple Coils (0x0f)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Write Multiple Registers (0x10)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QSpinBox" name="startAddr">
           <property name="maximum">
            <number>65535</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QListWidget" name="slaveID">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="isWrapping" stdset="0">
            <bool>true</bool>
           </property>
           <property name="viewMode">
            <enum>QListView::ListMode</enum>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="itemAlignment">
            <set>Qt::AlignLeading</set>
           </property>
           <property name="sortingEnabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>1</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>3</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>11</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>12</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>13</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>14</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>15</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
           <item>
            <property name="text">
             <string>16</string>
            </property>
            <property name="checkState">
             <enum>Checked</enum>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Registers</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QTableWidget" name="regTable">
           <property name="textElideMode">
            <enum>Qt::ElideLeft</enum>
           </property>
           <attribute name="horizontalHeaderCascadingSectionResizes">
            <bool>true</bool>
           </attribute>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>60</number>
           </attribute>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>60</number>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderDefaultSectionSize">
            <number>25</number>
           </attribute>
           <column>
            <property name="text">
             <string>Data type</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Register</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Data</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Explanation register</string>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="textAlignment">
             <set>AlignJustify|AlignVCenter</set>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Explanation data</string>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="textAlignment">
             <set>AlignJustify|AlignVCenter</set>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Bus Monitor</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Raw data received:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>201</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="clearBusRawData">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="4">
        <widget class="QPlainTextEdit" name="rawData">
         <property name="font">
          <font>
           <family>Fixedsys</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>4</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0" colspan="4">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="0" rowspan="2">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>4</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>ModBus requests/responses:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="2" rowspan="2">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>156</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="5" column="3" rowspan="2">
        <widget class="QPushButton" name="clearBusMonTable">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="4">
        <widget class="QTableWidget" name="busMonTable">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>true</bool>
         </attribute>
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>75</number>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>18</number>
         </attribute>
         <column>
          <property name="text">
           <string>I/O</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Slave ID</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Function code</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Start address</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Num of coils</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>CRC</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1021</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_MDF"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionBatchProcessing"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout_MDF">
   <property name="text">
    <string>About Modbus Data Fetcher</string>
   </property>
  </action>
  <action name="actionBatchProcessing">
   <property name="text">
    <string>Batch processing</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RtuSettingsWidget</class>
   <extends>QWidget</extends>
   <header>rtusettingswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TcpIpSettingsWidget</class>
   <extends>QWidget</extends>
   <header>tcpipsettingswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AsciiSettingsWidget</class>
   <extends>QWidget</extends>
   <header>asciisettingswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>dbForm</class>
   <extends>QWidget</extends>
   <header>dbform.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../data/qmodbus.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearBusRawData</sender>
   <signal>clicked()</signal>
   <receiver>rawData</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>833</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>669</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBatchProcessing</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>openBatchProcessor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openBatchProcessor()</slot>
 </slots>
</ui>
